packages/amplify-codegen/src/commands/models.js

```diff
@@ -99,19 +101,55 @@ function loadSchema(apiResourcePath) {
   throw new Error('Could not load the schema');
 }
 
-function getModelOutputPath(context) {
+async function getModelOutputPath(context) {
   const projectConfig = context.amplify.getProjectConfig();
+
+  if(projectConfig.modelgen && projectConfig.modelgen.outputPath && !context.parameters.options.configpath) {
+    context.print.success('\nThe output path for modelgen is already setup. You can edit the path in .config/project-config.json\n');
+    return projectConfig.modelgen.outputPath;
+  }
+
+  let defaultPath = '.';
   switch (projectConfig.frontend) {
     case 'javascript':
-      return 'src';
+      defaultPath = 'src/models';
+      break;
     case 'android':
-      return projectConfig.android && projectConfig.android.config && projectConfig.android.config.ResDir
+      defaultPath = projectConfig.android && projectConfig.android.config && projectConfig.android.config.ResDir
         ? path.normalize(path.join(projectConfig.android.config.ResDir, '..', 'java'))
         : path.join('app', 'src', 'main', 'java');
+      break;
     case 'ios':
-      return 'amplify/generated/models';
-    default:
-      return '.';
+      defaultPath =  'amplify/generated/models';
+      break;
+  }
+
+  let outputPath = defaultPath;
+  if(context.parameters.options.configpath) {
+    const answer = await askForModelOutputPath(defaultPath);
+    outputPath = answer.outputPath;
+  }
+
+  //Get the project root directory
+  let projectRoot;
+  try {
+    context.amplify.getProjectMeta();
+    projectRoot = context.amplify.getEnvInfo().projectPath;
+  } catch (e) {
+    projectRoot = process.cwd();
   }
+  context.amplify.updateProjectConfig(projectRoot, 'modelgen', {outputPath: outputPath});
+
+  return outputPath;
+}
+
+async function askForModelOutputPath(defaultPath) {
+  const answer = await inquirer.prompt({
+    type: 'input',
+    name: 'outputPath',
+    message: 'Please type your relative output path for model',
```

Comment:
The contents of this message is probably one of the most important things in this PR. Does it make sense to include an example?
```suggestion
    message: 'Enter the path to the directory where generated models will be output. The path should be relative to the project root (e.g. ./src/main/java/com/yourcompany/generated.)',
```

However, my message above is getting long.

Reply:
I use the message without content in parentheses. User can see the default path as an example when the question pops up.

Reply:
I include the examples in the PR description FYI.